Rubberduck.VBEEditor/Class1.cs

```diff
@@ -0,0 +1,964 @@
+
+            _currentTypeLibPtr = _currentTypeLibStruct.Next;
+            _currentTypeLibStruct = StructHelper.ReadStructureSafe<TypeLibObj_VBE>(_currentTypeLibPtr);
+            return true;
+        }
+    }
+
+    // the main class for hooking into the live ITypeLibs provided by the VBE
+    public class TypeLibsAccessor_VBE : DisposableList<TypeLibWrapper_VBE>, IDisposable
+    {
+        public TypeLibsAccessor_VBE(IVBE ide)
+        {
+            // We need at least one project in the VBE.VBProjects collection to be accessible (i.e. unprotected)
+            // in order to get access to the list of loaded project TypeLibs using this method
+
+            foreach (var project in ide.VBProjects)
+            {
+                using (project)
+                {
+                    try
+                    {
+                        using (var references = project.References)
+                        {
+                            // Now we've got the references object, we can read the internal object structure to grab the ITypeLib
+                            var internalReferencesObj = StructHelper.ReadStructure<ReferencesObj_VBE>(references.Target);
+
+                            // Now we've got this one internalReferencesObj.typeLib, we can iterate through ALL loaded project TypeLibs
+                            using (var typeLibIterator = new TypeLibsIterator_VBE(internalReferencesObj.TypeLib))
+                            {
+                                foreach (var typeLib in typeLibIterator)
+                                {
+                                    Add(typeLib);
+                                }
+                            }
+                        }
+
+                        // we only need access to a single VBProject References object to make it work, so we can return now.
+                        return;
+                    }
+                    finally
+                    {
+                        // probably a protected project, just move on to the next project.
+                    }
+                }
+            }
+
+            // return an empty list on error
+        }
+
+        public TypeLibWrapper_VBE FindTypeLib(string searchLibName)
+        {
+            foreach (var typeLib in this)
+            {
+                if (typeLib.Name == searchLibName)
+                {
+                    return typeLib;
+                }
+            }
+
+            throw new ArgumentException($"TypeLibsAccessor_VBE::FindTypeLib failed. '{searchLibName}' project not found.");
+        }
+    }
+
+    // for debug purposes, just reinventing the wheel here to document the major things exposed by a particular ITypeLib
+    // (compatible with all ITypeLibs, not just VBE ones, but also documents the VBE specific extensions)
+    // this is a throw away class, once proper integration into RD has been achieved.
+    public class TypeLibDocumenter
```

Comment:
My gut feeling is that we'll still need this even after we've integrated this PR. At every least, we could put it in a `Debug` folder one level down to be used.

Reply:
sure, it might be useful.  It works generically on all ITypeLibs, so probably doesn't belong here in this group

Reply:
Hmm. The other usage is in the [`Parsing.ComReflection`](https://github.com/rubberduck-vba/Rubberduck/tree/6d652422983af159c95efab9a544ececc72371ee/Rubberduck.Parsing/ComReflection) project.... But I _think_ that project can take a reference from here so no problem.